graylog2-web-interface/src/views/logic/views/ViewTransformer.js

```diff
@@ -0,0 +1,42 @@
+// @flow strict
+import { List, Map } from 'immutable';
+import Widget from 'views/logic/widgets/Widget';
+
+import type { ViewStateMap } from './View';
+import View from './View';
+import ViewState from './ViewState';
+
+const ViewTransformer = (searchView: View): View => {
+  const newViewStateMap: ViewStateMap = (searchView.state || Map.of()).map((viewState: ViewState, queryId: string) => {
+    const { timerange, query, filter = List.of() } = searchView.search.queries.find(q => q.id === queryId);
```

Comment:
We could prepare a `Map<QueryId, Query>` one level up once, so we can do a cheap lookup here instead of iterating over the list.